### PR TITLE
Use conditional build in rpm to manage dev-only settings

### PR DESCRIPTION
**Pull request by rocodes**
## Status

Work in progress  
(needs testing, needs rpmlinting and a fixup in `build-rpm.sh` if we decide to pursue this route - this is a napkin sketch) 

## Description of Changes
Refs #274 

To further the goal of removing more logic from Salt and into packaging (easier versioning, faster dom0 provisioning therefore faster updater runs, less salt breakage) we are moving files from being provisioned during the apply run into being packaged by the dom0 config and deb vmconfig packages (see eg #1030).

This is straightforward when files are static, but is somewhat trickier when files are either dynamically generated based on runtime info, or when they are provisioned conditionally (ie depending on build flavour).

Filing this PR as a discussion starting point for handling the latter situation. It introduces a [build condition](https://rpm-software-management.github.io/rpm/manual/conditionalbuilds.html) `powersettings` , which is on by default and currently configures the logind.conf power lid settings, but if we like this approach, should be expanded to include the power management xfce logic.

The goal of this type of change is to provide a path for deprecatating all* the the `if d.environment == "staging" or d.environment == "prod"`  type dynamic checking in our salt runs. *(All, or all but the keyring/repo provisioning, which will be handled in a separate bootstrap package.)

Changes proposed in this pull request:
* Replace inline configuration of logind.conf with a [drop-in file + directory](https://www.freedesktop.org/software/systemd/man/latest/systemd-system.conf.html) `/etc/systemd/login.conf.d/override_logind.conf`
* ...That is provisioned on prod and staging rpm builds (default option)
* ... And disabled on dev rpm builds (explicit build parameter passed to rpmbuild)
* And updates make targets and CI accordingly
Note: this change means that our power lid settings are applied when installing the RPM, as opposed to after successful orchestration. Per conversation with @zenmonkeykstop this seems like an acceptable decision given the amount of control we take over the SDW system, whether or not we go the conditional RPM build approach.

### More discussion
Another option could be to build fully separate rpm packages (workstation-dom0-config-staging, dom0-config-dev, etc). This could be a good idea particularly for rpm repo configuration (although in that case we might build different keyring/bootstrapping packages rather than different dom0 config packages. In any case, this solution is not mutually exclusive of that idea, but it's a little less complex.   

## Testing

- [ ] Visual Review
- [ ] CI Passes
- [ ] staging/ prod rpms are built correctly, and when installed, install a directory `/etc/systemd/login.conf.d/` with a configuration file that overrides power lid settings  
- [ ] Dev RPM is built correctly (in CI and locally) and does not include these changes

## Deployment

Any special considerations for deployment? Consider both:

1. Upgrading existing pilot instances
2. New installs

### If you have made changes to the provisioning logic

- [ ] All tests (`make test`) pass in `dom0`

### If you have added or removed files

- [ ] I have updated `MANIFEST.in` and `rpm-build/SPECS/securedrop-workstation-dom0-config.spec`

### If documentation is required

- [ ] I have opened a PR in the [docs repo](https://github.com/freedomofpress/securedrop-workstation-docs) for these changes, or will do so later
- [ ] I would appreciate help with the documentation

### CODE REVIEW

**Comment by rocodes** [comment]
Mostly filing this as a proof of concept and would love @legoktm 's opinion on this or other approaches

File: Makefile [hunk 1]
```diff
@@ -32,6 +32,11 @@ dev staging: assert-dom0 ## Configures and builds a dev or staging environment
 build-rpm: ## Build RPM package
 	USE_BUILD_CONTAINER=true $(CONTAINER) ./scripts/build-rpm.sh
 
+# Dev RPMs don't contain power settings adjustment (lid)
```
Comment (zenmonkeykstop) [inline Makefile:35]: IMO the build process should produce env-neutral packages, with env-specific stuff being set up at install- or run-time.

File: dom0/sd-clean-all.sls [hunk 2]
```diff
@@ -99,12 +99,6 @@ sd-cleanup-etc-changes:
     - repl: ''
     - backup: no
 
-{% if d.environment == "prod" or d.environment == "staging" %}
```
Comment (zenmonkeykstop) [inline dom0/sd-clean-all.sls:102]: Do we even need a conditional on this if we keep it? Is there a downside to the restart on dev environments?

File: files/logind_override.conf [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[Login]
```
Comment (zenmonkeykstop) [inline files/logind_override.conf:1]: This is definitely a better way to handle the logind config change than blockreplacing!

**Review by zenmonkeykstop — CHANGES_REQUESTED** [review]
I'm a fan of the override as a way to reduce complexity here but I don't think dev-only RPM builds are warranted. As it stands the only difference between dev and prod RPMs are signatures and version strings - IMO this reduces the chance of untested changes getting shipped to prod.

At the same time I do support removing environment-specific config overall from the RPM - one other thing to consider here would  be splitting dom0/sd-default-config.{yml,sls} out into env-specific versions and only packaging the prod version. To do dev or staging builds devs need to clone the repo to dom0, so the non-prod versions would be available and could be subbed in after installing the RPM but before doing the sdw-admin --apply step.

**Comment by rocodes** [comment]
Thanks everyone - so what I'll do is close this, submit a PR for the logind stuff, and we can prioritize systemd changes as the way to manage dev vs staging/prod power settings as @legoktm suggests.

**Pull request closed without merging**